- created cron handler to run weekly tasks email based user preference

khoded opened the pull request:
-implemented latest tasks by user preference  email template handler
-added template support for en and br
-initialized cron in server

## Description

> Description of the pull request
 a weekly Cronjob that will match users with emails about the latest task that are based on their skills preference and the task labels.
## Changes

## Issue
#558  

> Closes #558 

> Main page

## Steps to test
npm run start: dev
Steps needed to reproduce the scenario from this change to validate if the pull request solve this issue

- Create / log user
- Create Task
- assign a label to task
- assign skills to the user preference
- edit Cron time to test

alexanmtz commented on locales/result/br.json:
```diff
@@ -127,6 +127,8 @@
   "mail.task.bounties.title": "Ainda temos tarefas abertas com recompensas pagas no Gitpay que você pode se candidatar para resolver. Está achando o valor baixo? Você pode demonstrar interesse e fazer uma oferta!",
   "mail.task.bounties.action": "Se você quer contribuir com software livre, veja as tarefas que temos no Gitpay",
   "mail.task.bounties.calltoaction": "Veja as tarefas e envie seu interesse em resolvê-la",
+  "mail.task.interest.subject": "Temos problemas em aberto para você com base em suas preferências no Gitpay",
+  "mail.task.interest.title": "Existem problemas abertos com base na sua preferência que precisam ser resolvidos no Gitpay. Você pode mostrar seu interesse, fazer uma oferta ou contribuir para melhorar sua experiência e tentar novos desafios",
```
@khoded please do not change the result files

alexanmtz commented on locales/result/en.json:
```diff
@@ -127,6 +127,8 @@
   "mail.task.bounties.title": "There's still open issues with bounties on Gitpay. Apply to solve it. You think the bounty is low? You can make an offer",
   "mail.task.bounties.action": "Check our issues on Gitpay",
   "mail.task.bounties.calltoaction": "See the issue on Gitpay and apply to solve it",
+  "mail.task.interest.subject": "We have open issues for you based on your preferences on Gitpay",
+  "mail.task.interest.title": "There's open issues based on your preference that needs to be solved on Gitpay. You can show your interest, make an offer or contribute to improve your experience and try new challenges",
```
Here too @khoded 

These files are generated by our translation tool after we merge the solution

khoded commented:
Alright will update the PR asap and make the changes, thanks

alexanmtz commented:
@khoded can you provide automated tests of your Cron Job created?

This is a situation not easy to test manually, so we have automated tests to show them in action.

We have some examples here:
https://github.com/worknenjoy/gitpay/blob/master/test/cron.test.js

You should create tasks with labels and some user with preferences and covered scenarios where's there's no match and with match

khoded commented:
Awesome, I've been looking to do this, will look into the example and add the tests.

khoded commented:
- increased the limit to enable more tasks to be added

- added 'latest' template id  so more tasks can be displayed in the email, are we creating a new sendgrid in  template d-uuid the future  for  this type of cron or stick with latest task.

- refactored mail/task.js to only send task with matched with each user preference else skip the task and user email.

- noticed label contains array of name objects in the tests, refactored to flat array of names to fit regexp labelNamesarray.test(user.skills) condition

will be adding the automated tests next

alexanmtz commented on modules/mail/task.js:
```diff
@@ -239,25 +242,27 @@ ${i18n.__('mail.messageAuthor.message', { message })} <p>${Signatures.sign(langu
-          reason: i18n.__('mail.task.reason'),
-          subject: i18n.__('mail.task.interest.subject')
-        } })
+      if (tasks.length !== 0) {
```
maybe 
```suggestion
      if (tasks.length > 0) {
```
?

khoded replied:
yeah committed the changes

alexanmtz commented:
Great @khoded, I'm looking forward to seeing the automated tests
